### PR TITLE
Update ruby-smart-proxy-remote-execution-ssh to 0.4.1

### PR DESCRIPTION
**Pull request by adamruzicka**
- Update smart_proxy_remote_execution_ssh to 0.4.1
- Deprecate foreman-remote-execution-core
- Drop foreman-remote-execution-core completely

### CODE REVIEW

**Comment by adamruzicka** [comment]
**Also drops foreman_remote_execution_core completely**

File: plugins/smart_proxy_remote_execution_ssh/control [hunk 1]
```diff
@@ -11,7 +11,8 @@ Package: ruby-smart-proxy-remote-execution-ssh
 Architecture: all
 XB-Ruby-Versions: ${ruby:Versions}
 Depends: ${shlibs:Depends}, ${misc:Depends}, ruby | ruby-interpreter, foreman-proxy (>= 1.11.0~rc3),
-  ruby-foreman-remote-execution-core (>= 1.0.0),
-  ruby-smart-proxy-dynflow (>= 0.1.0), ruby-smart-proxy-dynflow (<< 1.0.0), ruby-net-ssh
+  ruby-smart-proxy-dynflow (>= 0.5.0), ruby-smart-proxy-dynflow (<< 1.0.0), ruby-net-ssh
+Breaks: ruby-foreman-remote-execution-core (<<1.5.0)
+Replaces: ruby-foreman-remote-execution-core (<<1.5.0)
```
Comment (evgeni) [inline plugins/smart_proxy_remote_execution_ssh/control:16]: 1.5.0 is the version that this PR is removing, so `<<` seems wrong? Maybe `<<1.5.1`?

**Review by evgeni — COMMENTED** [review]
So on Debian, the situation is a tad different (due to the way we vendor gems there).

If you only install Foreman with REX, no proxy whatsoever, you end up with the following:

```
root@27a56775b9c6:/# dpkg -l |grep foreman
ii  foreman                       2.5.1-1                amd64        Systems management web interface
ii  ruby-foreman-deface           1.5.3-1                all          Foreman Deface Plugin Dependency
ii  ruby-foreman-remote-execution 4.5.0-1                all          Foreman Remote Execution Plugin
ii  ruby-foreman-tasks            4.1.2-1                all          Tasks management engine for Foreman.
```

So no core is installed, nothing needs to be replaced. Nice.

Now you install `ruby-smart-proxy-remote-execution-ssh` and end up with:

```
root@27a56775b9c6:/# dpkg -l |grep foreman
ii  foreman                               2.5.1-1                amd64        Systems management web interface
ii  foreman-proxy                         2.5.1-1                all          RESTful proxies for DNS, DHCP, TFTP, and Puppet
ii  ruby-foreman-deface                   1.5.3-1                all          Foreman Deface Plugin Dependency
ii  ruby-foreman-remote-execution         4.5.0-1                all          Foreman Remote Execution Plugin
ii  ruby-foreman-remote-execution-core    1.4.2-1                all          Foreman remote execution - core bits
ii  ruby-foreman-tasks                    4.1.2-1                all          Tasks management engine for Foreman.
ii  ruby-foreman-tasks-core               0.3.5-1                all          Foreman tasks - core bits
```

At this point we could try the upgrade (via the staging repo of this PR) that should clean it out:

```
root@27a56775b9c6:/# apt install ruby-smart-proxy-remote-execution-ssh 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following package was automatically installed and is no longer required:
  ruby-net-scp
Use 'apt autoremove' to remove it.
The following additional packages will be installed:
  ruby-smart-proxy-dynflow
The following packages will be REMOVED:
  ruby-foreman-remote-execution-core
The following packages will be upgraded:
  ruby-smart-proxy-dynflow ruby-smart-proxy-remote-execution-ssh
2 upgraded, 0 newly installed, 1 to remove and 18 not upgraded.
Need to get 0 B/34.0 kB of archives.
After this operation, 41.0 kB of additional disk space will be used.
Do you want to continue? [Y/n] 
```

Interestingly this doesn't trigger when I try to just `apt dist-upgrade`, stupid apt…

**Comment by evgeni** [comment]
As far as I can tell, this is correct, but somehow `apt` disagrees with me :(

**Comment by evgeni** [comment]
:cursing_face: turns out `apt` was confused by an unrelated package in Adams staging archive.

**Review by evgeni — APPROVED** [review]

**Pull request merged**